app/controllers/observations_controller/new_and_create.rb

```diff
@@ -104,6 +104,7 @@ def create
     success = false unless validate_name(params)
     success = false unless validate_place_name(params)
     success = false unless validate_object(@observation)
+    success = false unless validate_project_checkboxes!
```

Comment:
Curious why the '!' at the end?  I think of that as meaning that the method is dangerous like raising an exception like update! or save! or has significant side effects (foo.downcase! vs bar = foo.downcase).  While technically this method changes the object by potentially setting some ivars, that seems like business as usual for a controller.  For example, most CRUD methods set ivars and don't have '!'s.  I guess when I see a token ending in '!' I expect there to be a "safer" method with the same name and no '!'.  There's an interesting discussion here: https://stackoverflow.com/questions/612189/why-are-exclamation-marks-used-in-ruby-methods#612196 that talks about both perspectives.